pkg/kubelet/network/hairpin/hairpin.go

```diff
@@ -41,15 +41,9 @@ var (
 )
 
 func SetUpContainer(containerPid int, containerInterfaceName string) error {
-	e := exec.New()
-	return setUpContainerInternal(e, containerPid, containerInterfaceName)
-}
-
-func setUpContainerInternal(e exec.Interface, containerPid int, containerInterfaceName string) error {
-	hostIfName, err := findPairInterfaceOfContainerInterface(e, containerPid, containerInterfaceName)
+	hostIfName, err := findPairInterfaceOfContainerInterface(exec.New(), containerPid, containerInterfaceName)
 	if err != nil {
-		glog.Infof("Unable to find pair interface, setting up all interfaces: %v", err)
-		return setUpAllInterfaces()
+		return err
```

Comment:
I think this will break hairpin on platforms that don't have ethtool installed. We should enforce that as a common requirement like iptables, before removing this.